cmd/arcaflow/main.go

```diff
@@ -57,53 +58,42 @@ func main() {
-	flag.StringVar(
-		&configFile,
-		"config",
-		configFile,
-		"The Arcaflow configuration file to load, if any.",
-	)
-	flag.StringVar(
-		&input,
-		"input",
-		input,
-		"The workflow input file to load. May be outside the workflow directory. If no input file is passed, "+
-			"the workflow is assumed to take no input.",
-	)
-	flag.StringVar(
-		&dir,
-		"context",
-		dir,
-		"The workflow directory to run from. Defaults to the current directory.",
-	)
-	flag.StringVar(
-		&workflowFile,
-		"workflow",
-		workflowFile,
-		"The workflow file in the current directory to load. Defaults to workflow.yaml.",
+	const (
+		helpUsage    = "Display this help output."
+		versionUsage = "Print Arcaflow Engine version and exit."
+		configUsage  = "The Arcaflow configuration file to load, if any."
+		inputUsage   = `The workflow input file to load. May be outside
+                           the workflow directory. If no input file is
+                           passed, the workflow is assumed to take no input.`
+		contextUsage = `The workflow directory to run from. Defaults 
+                           to the current directory.`
+		workflowUsage = `The workflow file in the current directory to load.
+                           Defaults to workflow.yaml.`
 	)
+	flag.BoolVar(&printVersion, "version", printVersion, versionUsage)
+	flag.BoolVar(&printVersion, "v", printVersion, versionUsage)
+	flag.StringVar(&configFile, "config", configFile, configUsage)
+	flag.StringVar(&configFile, "c", configFile, configUsage)
+	flag.StringVar(&input, "input", input, inputUsage)
+	flag.StringVar(&input, "i", input, inputUsage)
+	flag.StringVar(&dir, "context", dir, contextUsage)
+	flag.StringVar(&dir, "x", dir, contextUsage)
+	flag.StringVar(&workflowFile, "workflow", workflowFile, workflowUsage)
+	flag.StringVar(&workflowFile, "w", workflowFile, workflowUsage)
+
 	flag.Usage = func() {
 		_, _ = os.Stderr.Write([]byte(`Usage: arcaflow [OPTIONS]
-
+		
 The Arcaflow engine will read the current directory and use it as a context
 for executing the workflow.
 
 Options:
-
-  -version            Print the Arcaflow Engine version and exit.
-
-  -config FILENAME    The Arcaflow configuration file to load, if any.
-
-  -input FILENAME     The workflow input file to load. May be outside the
-                      workflow directory. If no input file is passed,
-                      the workflow is assumed to take no input.
-
-  -context DIRECTORY  The workflow directory to run from. Defaults to the
-                      current directory.
-
-  -workflow FILENAME  The workflow file in the current directory to load.
-                      Defaults to workflow.yaml.
+  -h, --help               ` + helpUsage + `
+  -v, --version            ` + versionUsage + `
+  -c, --config FILENAME    ` + configUsage + `
+  -i, --input FILENAME     ` + inputUsage + `
+  -x, --context DIRECTORY  ` + contextUsage + `  
```

Comment:
I don't really like `-x` for `--context`.  How about using `-C` for `--config` so that we can use `-c` for `--context` (or vice versa).

Reply:
That seems a little more confusing than `-x` for context, which has a clear X sound and character. But if all of you prefer that, I won't stop you.

Reply:
And, sometimes, when you have to start scrambling for alternative non-obvious "short" names, maybe it's just not worth trying to shorten them in the first place...

Another option would be to drop the semblance of traditional UNIX one-character option names (which these *aren't* anyway, unless we replace `flag`) and, if you really feel that "config" and "context" and "input" are impractically long to type, how about `--in`, `--ctx` and `--cfg` as "short" aliases that aren't (quite) so obscure? 😀

Reply:
Right.  If we're going to abbreviate the names, using something more mnemonic than a middle letter which represents nearly the last sound in the long-name would be good.  😀  As Dave points out, the cultural imperative for single letter option specifiers is really weak here, since, with this parser, we cannot combine them under the same single dash, so, if we want short aliases, we might as well indulge and give them a couple of letters.

That said, I would be on board with dropping the single-letter option specifiers and instead investing some effort into selecting good defaults for the the options -- that way, we don't even have to use the long names for the typical cases.  (I know that @mfleader has been tempted along these lines, as well....)